c/parallel/src/util/types.h

```diff
@@ -25,87 +25,47 @@ std::string cccl_type_enum_to_name(cccl_type_enum type, bool is_pointer = false)
 {
   std::string result;
 
-  if (is_pointer)
+  switch (type)
   {
-    switch (type)
-    {
-      case cccl_type_enum::INT8:
-
-        check(nvrtcGetTypeName<::cuda::std::int8_t*>(&result));
```

Comment:
Nader, your change looks good to me, especially because it passes all tests.

However, it does change the behavior, therefore it might be good to ask @gevtushenko (original author) specifically:

This approach

```
        check(nvrtcGetTypeName<::cuda::std::int8_t*>(&result));
```

was introduced with PR #2256 (in c/src/reduce.cu at the time):

https://github.com/NVIDIA/cccl/pull/2256/files#diff-2a8594900c7245f7952fc68f69ca42c59a016e5f8555d126b8f9168e421289a4R72-R187

What was the original intent?

I'm totally speculating here:

Do we maybe want to keep both functions, maybe rename this one to `cccl_type_enum_to_nvrtc_type_name()`, and adding the `bool is_pointer = false` argument to the other one? (But then again, all tests pass...)

Reply:
I spoke with Georgii and he said that having two functions was an oversight on his part, and that this change is fine

Reply:
Perfect, thanks!